swift/WorkflowTesting/Sources/RenderExpectations.swift

```diff
@@ -99,6 +102,14 @@ public struct ExpectedWorker {
     }
 }
 
+public struct ExpectedSideEffect {
```

Comment:
I think right now we don’t have a way to just send a `WorkflowAction` to `RenderTester`. So we can’t write tests that act like a side-effect did something (and we couldn’t convert `ExpectedWorker` to be built on `ExpectedSideEffect`. We should consider adding an action to `ExpectedSideEffect` or a way to send an action to `RenderTester` directly.

Reply:
Good point @bencochran . 

> or a way to send an action to RenderTester directly.

Any ideas on the best way to do that? 
* Pass in `context`, so that we can `makeSink()` and send event.
* Pass in `inout state` so that we can just modify the state directly.
* Add a generic `ActionType` to `ExpectedSideEffect` and pass in a `sink` of `ActionType`.

Reply:
@bencochran, @AquaGeek: Please take a look at `ExpectedSideEffect` when you get a chance. I've made it take in a closure which accepts the context.

Currently, the only `init` takes in `key` and `action`, however, I wanted to make sure we can pass in arbitrary block of code in the future to ensure it's in line with the `runSIdeEffect` api.